modules/lambdas/main.tf

```diff
@@ -193,58 +189,33 @@ resource "aws_lambda_function" "copy_files_to_archive" {
 
   environment {
     variables = {
-      PREFIX                = var.prefix
-      DATABASE_PORT         = var.database_port
-      DATABASE_NAME         = var.database_name
-      DATABASE_USER         = var.database_app_user
       COPY_RETRIES          = var.orca_recovery_retry_limit
       COPY_RETRY_SLEEP_SECS = var.orca_recovery_retry_interval
+      DB_QUEUE_URL          = var.orca_sqs_status_update_queue_id
     }
   }
 }
 
+resource "aws_lambda_event_source_mapping" "copy_files_to_archive_event_source_mapping" {
+  event_source_arn = var.orca_sqs_staged_recovery_queue_arn
+  function_name    = aws_lambda_function.copy_files_to_archive.handler
+}
 
 # Additional resources needed by copy_files_to_archive
 # ------------------------------------------------------------------------------
-# allow_s3_trigger - Permissions to allow S3 bucket trigger to invoke lambda
-resource "aws_lambda_permission" "allow_s3_trigger" {
-  # Create loop so we can handle multiple orca buckets
-  for_each = toset(local.orca_buckets)
-
+# Permissions to allow SQS trigger to invoke lambda
+# todo: Do we need to add a section for posting to the queues as well?
+resource "aws_lambda_permission" "allow_sqs_trigger" {
   ## REQUIRED
   action        = "lambda:InvokeFunction"
   function_name = aws_lambda_function.copy_files_to_archive.function_name
-  principal     = "s3.amazonaws.com"
+  principal     = "sqs.amazonaws.com"
 
   ## OPTIONAL
-  statement_id = "AllowExecutionFromS3"
-  source_arn   = "arn:aws:s3:::${each.value}"
+  statement_id = "AllowExecutionFromSQS"
+  source_arn   = module.sqs.orca_sqs_staged_recovery_queue_arn
 }
 
-
```

Comment:
Wondering if `resource "aws_lambda_permission" "allow_s3_trigger" ` and `resource "aws_s3_bucket_notification" "copy_lambda_trigger"` should be commented out instead of deleted since they will be needed with almost as is (diffrent lambda arn and name) when Rizbi finishes the task for the staged items. Thoughts?

Reply:
Seemed from previous experience that we preferred deleting over commenting, but that is coming up pretty soon... Added back in.

Reply:
Yep, I agree. Normally deletion is the way to go. was just thinking of saving some time since that lambda will be created soon.